### PR TITLE
<Expand />

### PR DESCRIPTION
**Pull request by densa**

### CODE REVIEW

**Comment by densa** [comment]
</img> </img> </img> </img>

File: __tests__/e2e/08_Expand.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import test from 'tape-async'
+import helper from 'tipsi-appium-helper'
+
+const { driver, select, idFromXPath } = helper
+
+test('<Expand />', async (t) => {
+  // await driver.pause(1000000)
```
Comment (isnifer) [inline __tests__/e2e/08_Expand.test.js]: Remove unused code

File: __tests__/e2e/08_Expand.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    android: idFromXPath(`//
+      android.widget.LinearLayout[1]/android.widget.FrameLayout[1]/
+      android.widget.FrameLayout[1]/android.view.ViewGroup[1]/
+      android.view.ViewGroup[2]/android.view.ViewGroup[1]/android.widget.ScrollView[1]
+    `),
+  })
+  const expandItemId = select({
+    ios: idFromXPath(`${expandGroupId}/XCUIElementTypeOther[2]/XCUIElementTypeOther`),
+    android: idFromXPath(`${expandGroupId}/android.view.ViewGroup[1]/
+                          android.view.ViewGroup[1]/android.view.ViewGroup[2]
```
Comment (isnifer) [inline __tests__/e2e/08_Expand.test.js]: try to make it in one line or divide into three line like: 
```javascript
android: idFromXPath(`
  ${expandGroupId}/android.view.ViewGroup[1]/android.view.ViewGroup[1]/android.view.ViewGroup[2]
`),
```

File: __tests__/e2e/08_Expand.test.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  try {
+    await helper.openExampleFor('<Expand />', 60000)
+    await driver.waitForVisible(expandItemId, 10000)
+    t.pass('<Expand /> example should be visible')
+
+    const height = await driver.getElementSize(expandItemId, 'height')
+    await driver.click(expandItemId)
+    t.pass('Can click on first expand item')
+    t.notEqual(height, await driver.getElementSize(expandItemId, 'height'), 'Height of the element should be changed after the click')
```
Comment (isnifer) [inline __tests__/e2e/08_Expand.test.js]: Please, install eslint plugin in your editor. This line more than 100 symbols. Edit line like this:
```js
const expandedHeight = await driver.getElementSize(expandItemId, 'height')
t.notEqual(
  height, 
  expandedHeight, 
  'Height of the element should be changed after the click'
)
```

File: src/Expand/Expand.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    const { title, description, children } = this.props
+    const { expanded } = this.state
+    const iconName = expanded ? 'chevron-up' : 'chevron-down'
+
+    return (
+      <TouchableOpacity onPress={this.onPress} activeOpacity={1}>
+        <View style={styles.container}>
+          <View style={styles.titleContainer}>
+            <Text numberOfLines={1}>
+              {title}
```
Comment (itsmepetrov) [inline src/Expand/Expand.js]: Title and description should use different styles

File: uiexplorer/examples/Expand.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            which surround their properties like gorgeous manicured lawns.
+            Other wineries don’t own vineyards but instead buy their grapes
+            from independent grape growers or buy juice or bulk wine from other
+            wine producers. And some Napa Valley wine producers don’t have
+            their own wineries; they bring their grapes to custom-crush wine
+            facilities, which they rent — all for the distinction of making “Napa Valley wine.”
+          </Text>
+        </View>
+        <Image
+          source={{ uri: 'https://dl.dropboxusercontent.com/u/1193220/Napa-Wine-Map-wine-folly.jpg' }}
```
Comment (itsmepetrov) [inline uiexplorer/examples/Expand.js]: use image from local storage

**Comment by itsmepetrov** [comment]
May be we can add expand/collapse animation via [LayouAnimation](https://facebook.github.io/react-native/docs/layoutanimation.html)

**Comment by itsmepetrov** [comment]
Blocked by #4

File: src/Expand/Expand.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    children: PropTypes.node,
+    defaultExpanded: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    expanded: false,
+  }
+  constructor(props) {
+    super(props)
+    this.state = { expanded: props.defaultExpanded }
```
Comment (itsmepetrov) [inline src/Expand/Expand.js]: move `state` to class params like:

```js
class Example {
  state = { test: 10 }
}
```

File: src/Expand/Expand.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  onPress = () => {
+    this.setState(
+      { expanded: !this.state.expanded }
+    )
```
Comment (itsmepetrov) [inline src/Expand/Expand.js]: rewrite to:

```js
this.setState({ expanded: !this.state.expanded })
```

**Comment by itsmepetrov** [comment]
Please provide better description in UIExplorer for this component

**Comment by itsmepetrov** [comment]
Please rebase

File: src/Expand/Expand.js (hunk 4 above)
Comment (densa) [inline src/Expand/Expand.js]: `<View style={styles.descriptionContainer}>`

**Comment by itsmepetrov** [comment]
Updated tests

**Comment by itsmepetrov** [comment]
Please update component for `themeable` HOC

File: README.md [hunk 8]
```diff
@@ -268,7 +293,6 @@ ThemeRegister.set({
     },
   },
 })
-```
```
Comment (itsmepetrov) [inline README.md]: Why?

File: README.md [hunk 9]
```diff
@@ -238,6 +238,31 @@ const Example = () => (
...
+#### Example
+```js
+import { Expand } from 'tipsi-ui-kit'
+
+<Expand
+  title="Winemakers Notes:"
+  description="The 2012 vintage in Napa Valley was about as close to ‘normal’ as it gets! "
+/>
+```
+#### Preview
```
Comment (itsmepetrov) [inline README.md]: Add info about styles structure

File: src/Expand/Expand.js (hunk 4 above)
Comment (isnifer) [inline src/Expand/Expand.js]: @densa I think @itsmepetrov meant that you should describe styles for Text too, cause we should provide a way to customize styles for this Text (color, fontSize, fontWeight, etc...)

File: src/Expand/Expand.js [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+    return (
+      <TouchableOpacity onPress={this.onPress} activeOpacity={1}>
+        <View style={styles.container}>
+          <View style={styles.titleContainer}>
+            <Text numberOfLines={1}>
+              {title}
+            </Text>
+            <Icon
+              name={iconName}
+              color="#9FADBA"
```
Comment (isnifer) [inline src/Expand/Expand.js]: I think this property should be customizable from the outside and should have default value from ThemeConstants // @itsmepetrov

File: uiexplorer/examples/Expand.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
+import React from 'react'
+import { Text, Image, View } from 'react-native'
+import register from '../core/utils/register'
+import Expand from '../../src/Expand'
+import Tags from '../../src/Tags'
```
Comment (isnifer) [inline uiexplorer/examples/Expand.js]: please, import Expand and Tags from `src`:
```js
import { Expand, Tags } from '../../src'
```

File: src/Expand/Expand.js (hunk 10 above)
Comment (itsmepetrov) [inline src/Expand/Expand.js]: Yes sure

**Comment by itsmepetrov** [comment]
Please replace example screens with following:

expand_ios
expand_android

**Comment by itsmepetrov** [comment]
We will merge this PR after #9

**Comment by itsmepetrov** [comment]
@densa please rebase

**Comment by itsmepetrov** [comment]
rename `__tests__/e2e/08_Expand.test.js` to `__tests__/e2e/09_Expand.test.js`

**Comment by itsmepetrov** [comment]
Looks good to merge

**Comment by cybergrind** [comment]
great

**Pull request merged**